source/engine/assets/NcAssetImpl.h

```diff
@@ -32,17 +33,20 @@ class NcAssetImpl : public NcAsset
 
         void OnBeforeSceneLoad() override;
 
+        auto OnBoneUpdate() noexcept -> Signal<const BoneUpdateEventData&>& override;
```

Comment:
My initial thoughts are that bones/meshes are not separate assets, therefore they (maybe?) shouldn't need different signals. I'm looking for the receiving code to better understand, but I'm only seeing the emit calls. Is there anything subscribing to this event in this PR, or is that coming down the pike?

Reply:
My personal preference is to leave it as two signals - we could combine it but I think it makes SkeletalAnimationSystem less clear, and we'd have to do "onSignalReceived() { if mesh.bonesData.has_value() ... } in the system as well, which I'd rather avoid